contracts/strategies/aave/StrategyAaveLendingPool.sol

```diff
@@ -0,0 +1,222 @@
+ */
+contract StrategyAaveLendingPool is IStrategy, Ownable {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    // The address of Aave Lending Pool
+    address public lendingPool;
+
+    // The address of supplying token (e.g. DAI, USDT)
+    address public supplyToken;
+
+    // The address of Aave interest-bearing token (e.g. aDAI, aUSDT)
+    address public aToken;
+
+    address public controller;
+
+    // The address of Aave Incentives Controller
+    address public incentivesController;
+
+    // The address of Aave StakedAave contract
+    address public stakedAave;
+
+    // The address of AAVE token
+    address public aave;
+
+    address public uniswap;
+    // The address of WETH token
+    address public weth;
+
+    uint256 internal constant MAX_INT = 2**256 - 1;
+    uint256 public shares;
+
+    constructor(
+        address _lendingPool,
+        address _supplyToken,
+        address _aToken,
+        address _controller,
+        address _incentivesController,
+        address _stakedAave,
+        address _aave,
+        address _uniswap,
+        address _weth
+    ) {
+        lendingPool = _lendingPool;
+        supplyToken = _supplyToken;
+        aToken = _aToken;
+        controller = _controller;
+        incentivesController = _incentivesController;
+        stakedAave = _stakedAave;
+        aave = _aave;
+        uniswap = _uniswap;
+        weth = _weth;
+    }
+
+    /**
+     * @dev Require that the caller must be an EOA account to avoid flash loans.
+     */
+    modifier onlyEOA() {
+        require(msg.sender == tx.origin, "Not EOA");
+        _;
+    }
+
+    function getAssetAddress() external view override returns (address) {
+        return supplyToken;
+    }
+
+    function aggregateOrders(
+        uint256 _buyAmount,
+        uint256 _sellShares,
+        uint256 _minSharesFromBuy,
+        uint256 _minAmountFromSell
+    ) external override returns (uint256, uint256) {
+        require(msg.sender == controller, "Not controller");
+        require(shares >= _sellShares, "not enough shares to sell");
+
+        // 1. Deposit or withdrawal
+        uint256 sharesFromBuy;
+        uint256 amountFromSell;
+        uint256 assetAmount = IAToken(aToken).balanceOf(address(this));
+        if (assetAmount == 0 || shares == 0) {
+            shares = _buyAmount;
+            assetAmount = _buyAmount;
+            sharesFromBuy = _buyAmount;
+        } else {
+            sharesFromBuy = _buyAmount.mul(shares).div(assetAmount);
+            amountFromSell = _sellShares.mul(assetAmount).div(shares);
+            assetAmount = assetAmount.add(_buyAmount).sub(amountFromSell);
+            shares = shares.add(sharesFromBuy).sub(_sellShares);
+        }
+        require(sharesFromBuy >= _minSharesFromBuy, "failed min shares from buy");
```

Comment:
These checks should happen after the interaction with Aave, not before.

Reply:
not an issue for aave